src/index.js

```diff
@@ -214,53 +203,74 @@ export default function undoable (reducer, rawConfig = {}) {
     jumpType: rawConfig.jumpType || ActionTypes.JUMP,
     clearHistoryType: rawConfig.clearHistoryType || ActionTypes.CLEAR_HISTORY
   }
-  config.history = rawConfig.initialHistory || createHistory(config.initialState || reducer(undefined, {}))
 
-  return (state = config.history, action = {}) => {
+  return (state, action = {}) => {
     debugStart(action, state)
+
+    let history
+    if (!config.history) {
+      debug('create history on init')
+
+      if (state === undefined) {
+        config.history = createHistory(reducer(state, {}))
+      } else if (isHistory(state)) {
+        config.history = state
+      } else {
+        config.history = createHistory(state)
+      }
```

Comment:
So we initialise the history here, when the reducer is called for the first time.